pkgs/build-support/setup-hooks/copy-desktop-items.sh

```diff
@@ -0,0 +1,42 @@
+# shellcheck shell=bash
+
+# Setup hook that installs specified desktop items.
+#
+# Example usage in a derivation:
+#
+#   { …, makeDesktopItem, copyDesktopItems, … }:
+#
+#   let desktopItem = makeDesktopItem { … }; in
+#   stdenv.mkDerivation {
+#     …
+#     nativeBuildInputs = [ copyDesktopItems ];
+#
+#     desktopItems =  [ desktopItem ];
+#     …
+#   }
+#
+# This hook will copy files which are either given by full path
+# or all '*.desktop' files placed inside the 'share/applications'
+# folder of each `desktopItems` argument.
+
+postInstallHooks+=(copyDesktopItems)
+
+copyDesktopItems() {
+    if [ "${dontCopyDesktopItems-}" = 1 ]; then return; fi
```

Comment:
```suggestion
    [[ ${dontCopyDesktopItems-} == 1 ]] && return
```
Normally in bash I would write the following but this looks more like sh or dash but you told shellcheck it is bash. I am a bit confused. Would you mind clearing this up?

Reply:
I took this syntax directly from another hook. There are many which use the same style, so I copied it:

https://github.com/NixOS/nixpkgs/blob/05d408e639f03be0c5db6e5c0d6621081f7106f2/pkgs/build-support/setup-hooks/move-lib64.sh#L11
https://github.com/NixOS/nixpkgs/blob/05d408e639f03be0c5db6e5c0d6621081f7106f2/pkgs/build-support/setup-hooks/move-sbin.sh#L8
https://github.com/NixOS/nixpkgs/blob/05d408e639f03be0c5db6e5c0d6621081f7106f2/pkgs/build-support/setup-hooks/update-autotools-gnu-config-scripts.sh#L4

I'm currently unaware if there is any policy about the hook sh/bash syntax, so I tried to stick with the existing hook style.

Reply:
I'm pretty sure both are valid bash. Isn't bash a superset of sh anyways?

Reply:
Use `[[` .. `]]` it's better behaved. And when in doubt see what shellcheck says hehe (though I think it will instruct on both).